api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1017,7 +1017,7 @@ public ConceptSource purgeConceptSource(ConceptSource cs) throws APIException {
 	@Override
 	public ConceptSource retireConceptSource(ConceptSource cs, String reason) throws APIException {
 		// retireReason is automatically set in BaseRetireHandler
-		return Context.getConceptService().saveConceptSource(cs);
```

Comment:
Did you change this accidentally?

Reply:
No, it's part of the ConceptSource validation changes

Reply:
Why this change?

Reply:
After changes in the `ConceptSourceValidator` using the `Context.getConceptService()` to access the `saveConceptSource()` while retiring the `ConceptSource` throws a hibernate exception. I think it's because the ConceptSource object trying to save is already retired by the `BaseRetireHandler` and can't be accessed.